tests/integration/integration-test.py

```diff
@@ -17,7 +17,16 @@ def setup(cls, config_path, openapi_path):
             cls.session = utils.setup_session(config)
             cls.test_cases = config['test_cases']
 
-        parser = ResolvingParser(openapi_path, backend='flex')
+        with open(openapi_path) as openapi_file:
+            openapi = yaml.load(openapi_file, Loader=yaml.SafeLoader)
+            if 'swagger' in openapi:
+                backend = 'flex'
+            elif 'openapi' in openapi:
+                backend = 'openapi-spec-validator'
```

Comment:
Clever. I think it will be nice to check if the corresponding package is also listed in the `requirement.txt` file.

Reply:
This would check if the package is in the required packages, but is it necessary to do this? Our requirements file includes both of these packages so this should always be true. If for some reason the package wasn't included, it would require a code change, and running the tests would cause a ModuleNotFoundError to be thrown so it should be obvious. We don't check if other packages like PyYAML exist in the requirements file before using them, so in what situations do you think we would need to make this check?

Reply:
@ianbrown9475  oh ok, my bad, I didn't notice both packages have been installed. Is this the default behavior when installing `prance`? I'm just not sure why we need to install both of them.

Reply:
prance doesn't install any of the validation packages. I added both to our requirements just so that either backend would work out of the box. Right now, our integration tests only work for swagger 2 documents but I'm assuming they will eventually work for either. So it's partly up to us to decide how the requirement dependencies will work. Either:

1. Both backend packages are included in the requirements. Only one will ever be used so there will be some wasted overhead of installing the packages that aren't required.
2. Neither backend packages are included in the requirements. The developer has to install the backend package depending on the openapi version being used. If being installed on Ansible, we will need to add some logic to determine this and install the extra packages. There will be some user/computational overhead for determining which backend to use and installing it. Our logic will also look slightly more complicated.

Thinking about it more, I think the logic in 2 could go in the docker file which could simplify this but I'd have to look into that more